### ISSUE TITLE
portable v1.30.2 autostart cannot load settings from .ini file

### ISSUE THREAD

**Issue by Hanamichi27** [FelixdelasPozas/TrayWeather#173]
as title, i'm using portable version.
Get my openweather api and setup and config settings.
I set to autostart at login but app cannot load settings and show as picture attacched.
If i close app and restart manually everything works correctly and load all my settings, position, api, etc.
What i'm doing wrong or what? 
Thanks for your app and everyone for helping
![Appunti_12-07-2024_01](https://github.com/user-attachments/assets/52100e42-db60-49d9-b823-bebce2aa21e8)
![Appunti_12-07-2024_02](https://github.com/user-attachments/assets/83e49958-a3cc-4e4d-9c00-0574a542ac38)

**Comment by FelixdelasPozas** [comment]
Hi @Hanamichi27.

I'm going to need additional data because this, as far as I know, never happened before to other users. 

- Windows version.
- Folder where TrayWeather is installed.
- In that folder, the file "TrayWeather.ini" is present? It has size 0?

Let's see if we can solve the issue.

Regards,
Félix.

**Comment by FelixdelasPozas** [comment]
I think I know where the error is in my code. I'll do some changes and more testing on this matter for the next release. When I release the next version I will update this issue to let you know.

Thanks!
Félix.

**Comment by FelixdelasPozas** [comment]
> Super! ITALIAN translation is ready for your revision.

Great! I'll add it to the next release. What name do you want me to put in the "About Dialog"? Hanamichi27?
For the name of the language I use the proper name in the language, in my case, Spanish is Español. What is the correct word for Italian?

Regards, 
Félix.

**Comment by FelixdelasPozas** [comment]
Hi @Hanamichi27 .

This issue should be fixed in the [latest release](https://github.com/FelixdelasPozas/TrayWeather/releases/tag/1.31.0).
Italian language has been added and can be selected in the configuration dialog. I'll close the issue if you confirm it to be solved.

**Comment by FelixdelasPozas** [comment]
Hi @Hanamichi27 .

Again, my mistake. There was a problem in the config dialog that prevented a correct initialization of some data if it was missing. 
Try with the one I just released [1.31.1](https://github.com/FelixdelasPozas/TrayWeather/releases/tag/1.31.1).

Hope this fixes it.
Regards, 
Félix.

**Comment by FelixdelasPozas** [comment]
I've just tried both providers with an empty INI file and in both cases it worked with no problems. Let me get this right, the application works if you execute it, but it doesn't work when put in the auto-start (executing when logging in). It executes but cannot find the INI file and so it opens the configuration dialog.

Am I missing something?

**Comment by FelixdelasPozas** [comment]
No, it's not needed to run it as administrator. It's an strange error, I'll look if there are any limitations to an application if executed in that mode. 
Usually the application looks into the folder it's been executed to find the ini file. Maybe when running from the registry other things apply. Currently I don't have any clue. I too use the autostart feature, but I'm using the registry.
 I'll look into it.

**Comment by FelixdelasPozas** [comment]
I've changed a few lines for the [latest release](https://github.com/FelixdelasPozas/TrayWeather/releases/tag/1.32.0). Please check if the latest changes solved the issue. Apparently the UI library don't check some NTFS permissions, hope that is the problem.

Regards, 
Félix.

**Comment by fhumayun** [comment]
Greetings I installed trayweather using scoop and have a similar situation where, upon restarting my workstation, the app launches but loses all the configuration settings I had previously setup.


![Image](https://github.com/user-attachments/assets/5d549703-8d1c-4c00-875c-ad9106aff77e)

**Comment by FelixdelasPozas** [comment]
I didn't know scoop even existed, I should invest some time with Chocolatey. Also the about dialog seems to scale leaving a lot of free space, that I suppose is because you have a bigger DPI than me, for comparison: 

![Image](https://github.com/user-attachments/assets/84daedc6-2694-4c6b-b50e-7da9ee1a70fc)

About the portability problem you have I'm still out of ideas. Usually when you have a bug it happens for everyone, and if not, its usually a "corner case" for that specific user. But it's still worth investigating and solving.

A few more questions:
- The folder is "E:\Utility[Portable] Tray Weather\" so the path to INI file is "E:\Utility[Portable] Tray Weather\TrayWeather.ini", right?
- Is the drive E local? I mean, it's not mounted using NFS.

The next relase will have more code and checks trying to solve this.

Regards,
Félix.

**Comment by fhumayun** [comment]
Yes, the E drive is an attached magnetic disk drive - I try to avoid cluttering my limited space NVMe ssd C drive.

But, I had much better luck with downloading an older release from [Softonic](https://tray-weather.en.softonic.com)

![Image](https://github.com/user-attachments/assets/e80d1ea0-baf3-4851-a14b-10b7a7372e12)

**Comment by FelixdelasPozas** [comment]
Ups @fhumayun I've confused you with Hanamichi27, that started the thread. Just a note, I would avoid Softonic as they were known to get free software from Github and "repackage it", sometimes with malware.

Apart from that, it's a useful hint that you're using an external USB, thanks!
